src/driver/hermes_mod.c

```diff
@@ -55,12 +55,102 @@ static const struct pci_device_id pci_ids[] = {
 };
 MODULE_DEVICE_TABLE(pci, pci_ids);
 
+static void irq_teardown(struct xdma_dev *xdev)
+{
+	irq_msix_channel_teardown(xdev);
+}
+
+static int irq_setup(struct xdma_dev *xdev)
+{
+	return xdma_irq_setup(xdev);
+}
+
+/*
+ * code to detect if MSI/MSI-X capability exists is derived
+ * from linux/pci/msi.c - pci_msi_check_device
+ */
+
+#ifndef arch_msi_check_device
+static int arch_msi_check_device(struct pci_dev *dev, int nvec, int type)
```

Comment:
This is highly questionable.... Why is the IRQ handling of this device so complicated and special?

Reply:
Hmm this might be a leftover of their support for ancient kernels. Or not. But anyway, you're right, this and msi_msix_capable() aren't required